### ISSUE TITLE
:set-mark and :jump-mark do only work with 100% zoom

### ISSUE THREAD

**Issue by m2habert** [qutebrowser/qutebrowser#4013]
I prefer to use 90% zoom. My marks never work, and the error / mismatch gets more obvious the bigger the page is.

I've just checked now and, well, if you go to 100% zoom and then you set the mark, it will work properly (if you stay in 100% zoom).

**Comment by jgkamat** [comment]
This seems to be because scrollPosition seems to ignore zoom when calculating
pixels currently, while js dosen't.

https://doc.qt.io/archives/qt-5.10/qwebenginepage.html#scrollPosition-prop

Ideally we could use the "unzoomed" pixel value, so that the marks would
continue working despite a zoom, but I'm not able to find an API for that in
js.

m2habert writes:

**Comment by m2habert** [comment]
I could be wrong, but I think that vimperator would store the percentage value. It is certainly not precise, but it ends up way closer to the mark.

**Comment by jgkamat** [comment]
I think this would be a bad idea because some websites dynamically add more content to the end of the page with js, which would break where marks take you.

there's probably a cleaner way of storing position but I'm not sure what it is. 

On June 23, 2018 7:30:21 PM EDT, m2habert <notifications@github.com> wrote:
>I could be wrong, but I think that vimperator would store the
>percentage value. It is certainly not precise, but it ends up way
>closer to the mark.

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by codesections** [comment]
This bug is also triggered if when the user is at 100% zoom but has `c.qt.hdpi = True` set, presumably for the same reasons.

**Comment by user202729** [comment]
Would it be easier to divide the obtained scroll position by the zoom factor?

**Comment by rien333** [comment]
> This bug is also triggered if when the user is at 100% zoom but has c.qt.hdpi = True set, presumably for the same reasons.

Yes, jump marks have been completely broken for me for quite a while now. I don't have `qt.highdpi = True` set, but I do have a hidpi screen which I compensate for with QT env variables, e.g. `QT_AUTO_SCREEN_SCALE_FACTOR` and the like. For some reason the bug persists when running `qutebrowser -T`, but that could be related to aforementioned variables.  It would be nice to regain the ability to use jump marks, until then I might cook up a quick hack using m2haberts' suggestion to use the scoll percentage value.

**Comment by Jumblemuddle** [comment]
This appears to be the case when using `QT_SCALE_FACTOR` alone as well.
I have `QT_SCALE_FACTOR=1.5` set, and if I set the `Zoom level` to `67%`, it appears to work as expected.

**Comment by brianbuccola** [comment]
Any update on whether/how this might be resolved? Or any kind of workaround? I've reached that point in life where I've `set zoom.default 125%` (:sob:). I don't set marks very often, but I *do* use `''` all the time.

**Comment by tonyd33** [comment]
I hope this gets fixed soon, but in the meantime, I find that a reasonable workaround not being able to set the zoom without breaking `set-mark`/`jump-mark` is to set `fonts.web.size.default` smaller/higher.
